File: src/main/java/core/db/test/Student.java
package core.db.test;

import core.db.annotation.*;

import java.util.List;

@Entity(table = "Student")
public final class Student {

    @Id(autoincrement = true)
    private long id;

    @Property(name = "first_name")
    private String firstName;

    @Property(name = "last_name")
    private String lastName;

    @Property(name = "year_of_study")
    private int yearOfStudy;

    @ToOne(joinedEntity = Supervisor.class)// todo remove useless type hint
    private Supervisor supervisor;

    @ToMany(joinedEntity = Lesson.class)// todo extract generic type instead
    private List<Lesson> lessons;

    @JoinEntity(joiningEntity = JoinTable.class, sourceProperty = "STUDENT_ID", targetProperty = "SOMETHING_ID")
    private List<Something> somethings;

}

File: src/main/java/core/db/test/Student.scala
package core.db.test

import java.util

import com.j256.ormlite.field.{DatabaseField => Column, ForeignCollectionField => ToMany}
import com.j256.ormlite.table.{DatabaseTable => Table}

object Student {
  def apply(): Student = new Student()
}

@Table(tableName = "Student")
final class Student private() {

  @Column(generatedId = true)
  var id: Int = _

  @Column(columnName = "first_name")
  var firstName: String = _

  @Column(columnName = "last_name")
  var lastName: String = _

  @Column(columnName = "year_of_study")
  var yearOfStudy: Int = _

  @Column(columnName = "supervisor",
    foreign = true,
    foreignAutoCreate = true,
    foreignAutoRefresh = true)
  var supervisor: Supervisor = _

  @ToMany(eager = true)
  var lessons: util.Collection[Lesson] = _

  override def toString: String =
    s"""id=$id,
       |firstName=$firstName lastName=$lastName,
       |year=$yearOfStudy, supervisor=$supervisor,
       |lessons=${new util.ArrayList[Lesson](lessons)}""".stripMargin
}
